common/djangoapps/third_party_auth/settings.py

```diff
@@ -85,3 +84,11 @@ def apply_settings(django_settings):
         'social.apps.django_app.context_processors.backends',
         'social.apps.django_app.context_processors.login_redirect',
     )
+
+    if not hasattr(django_settings, 'USER_FIELDS'):
```

Comment:
Can you please add some comments here explaining why the code is changing `USER_FIELDS` and explaining the new `FAKE_EMAIL_DOMAIN` and `THIRD_PARTY_AUTH_FALLBACK_FULL_NAME` settings?

Also, isn't this going to overwrite any custom value of `FAKE_EMAIL_DOMAIN` and `THIRD_PARTY_AUTH_FALLBACK_FULL_NAME` ? If those are supposed to be customizable by the user they should be added to lms/envs/common.py and lms/envs/aws.py, rather than this file.